receiver/snowflakereceiver/documentation.md

```diff
@@ -0,0 +1,84 @@
+[comment]: <> (Code generated by mdatagen. DO NOT EDIT.)
+
+# snowflakereceiver
+
+## Metrics
+
+These are the metrics available for this scraper.
+
+| Name | Description | Unit | Type | Attributes |
+| ---- | ----------- | ---- | ---- | ---------- |
+| **snowflake.billing.cloud_service.sum** | cloud services sum | credits | Gauge(Double) | <ul> <li>service_type</li> </ul> |
```

Comment:
Few comments for all the metrics:

1. Please make sure that all sum metrics are monotonic or non-monotonic sums, not gauges. So it's clear that they can be aggregated over the all the attributes by sum.
2. Please improve the descriptions: provide some idea about what the metric represents, not just repeating its name. And description should be a sentence starting with a capital letter.
3. Please don't make all the metrics enabled by default. Is seems to be a lot of data. As a rule of thumb, metrics enabled by default should provide critical information about heathiness and performance of a monitored system. Other metrics should be optional. All `snowflake.billing` metrics are good candidate for optional metrics. And maybe it's even better to start with a smaller set of essential metrics and add more later on?
4. Enclose all countable units into {brackets}, e.g. credits -> {credits}.

Reply:
Was on PTO going through this now! So sorry for the delay...

1. There is a naming conflict here. In the original snowflake monitoring tool the word sum is meant to represent the aggregate total as reported by the sql query which actually contains the results. This table entry is not aggregated by the observability tool, but rather displayed as it is reported. Should the name of these sum metrics be changed to reflect that or would keeping with previous convention be better?
2. Heard.
3. Heard. I was not initially aware that this was the norm, will fix this moving forwards.
4. Heard.

Reply:
I went ahead and changed all metrics named "sum" to "total" for clarity